gateways/stripe/stripe_vars.php

```diff
@@ -29,94 +32,21 @@ function espresso_display_stripe($payment_data) {
-					</p>
-					<p>
-						<label for="city"><?php _e('City', 'event_espresso'); ?></label>
-						<input name="city" type="text" id="stripe_city" class="required" value="<?php echo $city ?>" />
-					</p>
-					<p>
-						<label for="state"><?php _e('State', 'event_espresso'); ?></label>
-						<input name="state" type="text" id="stripe_state" class="required" value="<?php echo $state ?>" />
-					</p>
-					<p>
-						<label for="zip"><?php _e('Zip', 'event_espresso'); ?></label>
-						<input name="zip" type="text" id="stripe_zip" class="required" value="<?php echo $zip ?>" />
-					</p>
-				</fieldset>
 
-				<fieldset id="stripe-credit-card-info-dv">
-					<h4 class="section-title"><?php _e('Credit Card Information', 'event_espresso'); ?></h4>
-					<p>
-						<label for="card_num"><?php _e('Card Number', 'event_espresso'); ?></label>
-						<input type="text" name="cc" class="required" id="stripe_cc" autocomplete="off" />
-					</p>
-					<p>
-						<label for="card-exp"><?php _e('Expiration Month', 'event_espresso'); ?></label>
-						<select id="stripe_card-exp" name ="exp_month" class="med required">
-							<?php
-							$curr_month = date("m");
-							for ($i = 1; $i < 13; $i++) {
-								$val = $i;
-								if ($i < 10) {
-									$val = '0' . $i;
-								}
-								$selected = ($i == $curr_month) ? " selected" : "";
-								echo "<option value='$val'$selected>$val</option>";
-							}
-							?>
-						</select>
-					</p>
-					<p>
-						<label for="exp-year"><?php _e('Expiration Year', 'event_espresso'); ?></label>
-						<select id="stripe_exp_year" name ="exp_year" class="med required">
-							<?php
-							$curr_year = date("Y");
-							for ($i = 0; $i < 10; $i++) {
-								$disp_year = $curr_year + $i;
-								$selected = ($i == 0) ? " selected" : "";
-								echo "<option value='$disp_year'$selected>$disp_year</option>";
-							}
-							?>
-						</select>
-					</p>
-					<p>
-						<label for="cvv"><?php _e('CVC Code', 'event_espresso'); ?></label>
-						<input type="text" name="csc" id="stripe_csc" autocomplete="off" class="small required" />
-					</p>
-				</fieldset>
-				<input name="amount" type="hidden" value="<?php echo number_format($event_cost, 2) ?>" />
-				<input name="stripe" type="hidden" value="true" />
-				<input name="id" type="hidden" value="<?php echo $attendee_id ?>" />
-				<p class="event_form_submit">
-					<input name="stripe_submit" id="stripe_submit" class="submit-payment-btn allow-leave-page" type="submit" value="<?php _e('Complete Purchase', 'event_espresso'); ?>" />
-					<div class="clear"></div>
-				</p>
+			<form action="<?php echo add_query_arg(array('r_id'=>$registration_id, 'attendee_id'=>$attendee_id), get_permalink($org_options['return_url'])); ?>" method="POST" class="allow-leave-page">
+			  <script
+				src="https://checkout.stripe.com/v2/checkout.js" class="stripe-button"
```

Comment:
if reasonable, all scripts should be included in the plugin. See https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#8-plugins-may-not-send-executable-code-via-third-party-systems where it says "all non-service related JavaScript and CSS must be included locally". If not possible because their JS is dynamic or something, then we'll let it slide

Reply:
I don't think this rule applies here (this part of the plugin isn't included on the hosted version on wp.org) and EE4 Stripe's gateway uses the same script from Stripe:

https://github.com/eventespresso/eea-stripe-gateway/blob/master/forms/BillingForm.php#L175

Although the script should be enqueued (like how it's enqueued in the EE4 Stripe gateway).

Reply:
oups! ya good point @joshfeck

Reply:
We also can not include checkout.js within the plugin itself and keep the lowest PCI requirement levels.